node/src/tasks/user_sends_file.rs

```diff
@@ -201,211 +204,233 @@ where
+                let mut retry_attempts = 0;
+                loop {
+                    let upload_response = self
                         .storage_hub_handler
-                        .file_storage
-                        .read()
-                        .await
-                        .generate_proof(&file_key, &current_batch)
-                    {
-                        Ok(proof) => proof,
+                        .file_transfer
+                        .upload_request(peer_id, file_key.as_ref().into(), proof.clone(), None)
+                        .await;
+
+                    match upload_response {
+                        Ok(r) => {
+                            debug!(
+                                target: LOG_TARGET,
+                                "Successfully uploaded batch for file {:?} to peer {:?}",
+                                file_metadata.fingerprint,
+                                peer_id
+                            );
+
+                            // If the provider signals they have the entire file, we can stop
+                            if r.file_complete {
+                                info!(
+                                    target: LOG_TARGET,
+                                    "Stopping file upload process. Peer {:?} has the entire file {:?}",
+                                    peer_id,
+                                    file_metadata.fingerprint
+                                );
+                                return Ok(());
+                            }
+
+                            break;
+                        }
+                        Err(RequestError::RequestFailure(RequestFailure::Refused))
+                            if retry_attempts < 3 =>
+                        {
+                            warn!(
+                                target: LOG_TARGET,
+                                "Batch upload rejected by peer {:?}, retrying... (attempt {})",
+                                peer_id,
+                                retry_attempts + 1
+                            );
+                            retry_attempts += 1;
+
+                            // Wait for a short time before retrying
+                            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                        }
+                        Err(RequestError::RequestFailure(RequestFailure::Refused)) => {
+                            // If MSP doesn't receive file, the burden of downloading the file will be on the MSP.
+                            return Err(anyhow::anyhow!("Failed to send file {:?}", file_key));
+                        }
                         Err(e) => {
                             return Err(anyhow::anyhow!(
-                                "Failed to generate proof for batch of file {:?}\n Error: {:?}",
-                                file_key,
+                                "Failed to upload batch to peer {:?} (Error: {:?})",
```

Comment:
```suggestion
                                "Unexpected error while trying to upload batch to peer {:?} \n    Error: {:?}",
```

I think this error message is clearer, given that it an error happened for which we didn't have a way to handle it. Also, the line break to print the `Error` helps readability.

Reply:
Same than before. It just print the error code and doesn't have descriptive error message so I removed the break line so it not floating around in the log.